Header header
int32 n
Neighbor[] neighbors